### PR TITLE
[studio-ui] sidebar updates

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4351
https://github.com/craftercms/craftercms/issues/4358
https://github.com/craftercms/craftercms/issues/4353

### CODE REVIEW

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx [hunk 1]
```diff
@@ -85,12 +84,22 @@ const menuOptions = {
   }
 };
 
+interface OverrideClass {
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: StateStylingProps

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx [hunk 2]
```diff
@@ -124,10 +133,11 @@ export interface WidgetState {
 
 // PathNavigator
 export default function(props: WidgetProps) {
-  const { title, icon, path, id = removeSpaces(props.title), locale } = props;
+  const { label, icon, container, rootPath: path, id } = props;
   const state = useSelection((state) => state.pathNavigator)[id];
   const itemsByPath = useSelection((state) => state.content.items).byPath;
   const site = useActiveSiteId();
+  const locale = 'en';
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: Locale should still be a prop. If not supplied, use default locale

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx [hunk 3]
```diff
@@ -389,9 +401,29 @@ export function WidgetUI(props: any) {
...
-        icon={icon}
+        icon={
+          icon.baseClass
+            ? `${icon.baseClass} ${state?.collapsed ? icon.collapsedClass : icon.expandedClass}`
+            : icon
+        }
+        style={{
+          ...icon.baseStyle,
+          ...(state?.collapsed ? icon.collapsedStyle : icon.expandedStyle)
+        }}
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: - Is state ever undefined here? Perhaps we need a wrapping loading state or resource?
- If state is ever undefined, these conditions would always use the expanded style, which I'm not sure that's right

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 4]
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
...
-const SiteConfigLink = () => (
-  <LinkWithIcon label="Site Config" icon="fa fa-sliders" href="/studio/site-config" />
-);
-
 const ItemToComponentMap = {
-  PagesWidget: Widget,
-  'site-config': SiteConfigLink,
-  dashboard: DashboardLink
+  'craftercms.pathNavigator': Widget,
+  'craftercms.link': LinkWithIcon
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: craftercms.link => craftercms.linkWithIcon

File: ui/app/src/utils/xml.ts [hunk 5]
```diff
@@ -151,7 +151,7 @@ export function toJS(xml: string | XMLDocument): any {
   if (typeof xml !== 'string') {
     xml = serialize(xml);
   }
-  return parser.parse(xml, {});
+  return parser.parse(xml, { attributeNamePrefix:"", ignoreAttributes: false});
```
Comment (rart) [inline ui/app/src/utils/xml.ts]: - Format(!!!!!)
- Add options prop

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
Fix codacy

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx (hunk 3 above)
Comment (aachavarria) [reply ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: state is never undefined, this was my mistake. I will remove state validation.

File: ui/app/src/utils/xml.ts [hunk 6]
```diff
@@ -145,13 +148,13 @@ export function wrapElementInAuxDocument(element: Element): XMLDocument {
...
-export function toJS(xml: XMLDocument): any;
-export function toJS(xml: string | XMLDocument): any {
+export function toJS(xml: string, options?: X2jOptionsOptional): any;
+export function toJS(xml: XMLDocument, options?: X2jOptionsOptional): any;
+export function toJS(xml: string | XMLDocument, options?: X2jOptionsOptional): any {
   if (typeof xml !== 'string') {
     xml = serialize(xml);
   }
-  return parser.parse(xml, {});
+  return parser.parse(xml, options);
```
Comment (rart) [inline ui/app/src/utils/xml.ts]: Default these to the ones you had.
```js
{
  blah: blah,
  blah: blah,
  ...options
}
```

File: ui/app/src/utils/xml.ts [hunk 7]
```diff
@@ -145,13 +148,13 @@ export function wrapElementInAuxDocument(element: Element): XMLDocument {
   return fromString(`<?xml version="1.0" encoding="UTF-8"?>${element.outerHTML}`);
 }
 
-export function toJS(xml: string): any;
-export function toJS(xml: XMLDocument): any;
-export function toJS(xml: string | XMLDocument): any {
+export function toJS(xml: string, options?: X2jOptionsOptional): any;
+export function toJS(xml: XMLDocument, options?: X2jOptionsOptional): any;
```
Comment (rart) [inline ui/app/src/utils/xml.ts]: Overrides should go as
```js
export function toJS(xml: string): any;
export function toJS(xml: XMLDocument): any;
export function toJS(xml: string, options: X2jOptionsOptional): any;
export function toJS(xml: XMLDocument, options: X2jOptionsOptional): any;
export function toJS(xml: string | XMLDocument, options?: X2jOptionsOptional): any {
  // ..implementation
```

File: static-assets/components/cstudio-common/common-api.js [hunk 8]
```diff
@@ -897,7 +897,7 @@ var nodeOpen = false,
         });
       },
 
-      viewDependencies: function (site, items, approveType, defaultSelection) {
+      viewDependencies: function (site, items, approveType, defaultSelection, rootPath) {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Revert and extract from item

File: static-assets/components/cstudio-common/common-api.js [hunk 9]
```diff
@@ -906,6 +906,7 @@ var nodeOpen = false,
           payload: {
             open: true,
             item: CrafterCMSNext.util.content.parseLegacyItemToSandBoxItem(items[0]),
+            rootPath,
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Extract from item

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js [hunk 10]
```diff
@@ -1741,7 +1741,9 @@ var storage = CStudioAuthoring.Storage;
                       CStudioAuthoring.Operations.viewDependencies(
                         CStudioAuthoringContext.site,
                         selectedContent,
-                        false
+                        false,
+                        null,
+                        oCurrentTextNode.instance.config.params.path
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js]: Revert & extract from item

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx [hunk 11]
```diff
@@ -85,12 +85,23 @@ const menuOptions = {
...
-  path: string;
-  icon?: string | React.ElementType;
   id: string;
-  title?: string;
-  locale: string;
+  label: string;
+  rootPath: string;
+  excludes: string[];
+  locale?: string;
+  icon: StateStylingProps | React.ElementType;
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: `icon: Partial< StateStylingProps>;`

File: ui/app/src/components/Navigation/PathNavigator/Widget.tsx [hunk 12]
```diff
@@ -399,23 +420,43 @@ export function WidgetUI(props: any) {
...
+        ...container.baseStyle,
+        ...(state.collapsed ? container.collapsedStyle : container.expandedStyle)
+      }}
+    >
       <Header
-        icon={icon}
+        icon={
+          icon.baseClass
+            ? `${icon.baseClass} ${state.collapsed ? icon.collapsedClass : icon.expandedClass}`
+            : icon
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/Widget.tsx]: Fix logic

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 13]
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
...
+  altText: string;
+  link: string;
+  icon: {
+    baseClass: string;
+    baseStyle: object;
+  };
+}
+
+const LinkWithIcon = (props: LinkWithIconProps) => {
+  const { link, label, altText, icon } = props;
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: Alt text not applicable to links

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 14]
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
   })
 );
 
-const LinkWithIcon = ({ label, icon, href }) => {
+interface LinkWithIconProps {
+  label: string;
+  altText: string;
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: x

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 15]
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
     window.location.href = `${authoringBase}/preview#/?page=${computedUrl}&site=${site}`;
   }, [authoringBase, computedUrl, site]);
 
+  // region UI Config
+  useEffect(() => {
+    if(site){
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: Format

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 16]
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
...
+      dispatch(fetchSiteUiConfig({ site }))
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: Semi

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 17]
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
...
+    }
+  }, [dispatch, site])
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: semi

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 18]
```diff
@@ -14,30 +14,23 @@
...
  */
 
-import React, { useEffect } from 'react';
+import React from 'react';
 import { defineMessages, useIntl } from 'react-intl';
 import { createStyles } from '@material-ui/core';
 import makeStyles from '@material-ui/core/styles/makeStyles';
 import ToolPanel from './ToolPanel';
-import { SidebarConfigItem } from '../../../services/configuration';
+import { siteExplorerItem } from '../../../services/configuration';
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: Interface name should be capitalised

File: ui/app/src/services/configuration.ts [hunk 19]
```diff
@@ -28,17 +28,24 @@ import ContentType, { ContentTypeField } from '../models/ContentType';
...
-    map(({ response }) => response.content)
-  );
+export function getRawConfiguration(
+  site: string,
+  configPath: string,
+  module: CrafterCMSModules
+): Observable<string> {
+  return get(
+    `/studio/api/2/configuration/get_configuration?siteId=${site}&module=${module}&path=${configPath}`
+  ).pipe(map(({ response }) => response.content));
```
Comment (rart) [inline ui/app/src/services/configuration.ts]: use pluck(response, content)

File: ui/app/src/services/configuration.ts [hunk 20]
```diff
@@ -62,14 +69,14 @@ const LegacyPanelIdMap: any = {
   'ice-tools-panel': 'craftercms.ice.ice',
   'component-panel': 'craftercms.ice.components',
   'medium-panel': 'craftercms.ice.simulator',
-  'targeting': 'craftercms.ice.audiences'
+  targeting: 'craftercms.ice.audiences'
```
Comment (rart) [inline ui/app/src/services/configuration.ts]: let’s talk about removing all these mappings, being a brand new config, we don’t need to map legacy stuff

File: ui/app/src/services/configuration.ts [hunk 21]
```diff
@@ -258,67 +197,77 @@ function deserializeActiveTargetingModelData<T extends Object>(data: T, contentT
...
+            sidebar: {
+              panels: panelsParser(parsed.preview.sidebar.panels)
+            },
+            siteNav: {
+              links: linksParser(parsed.preview.siteNav.links)
+            }
+          }
+        };
       } else {
         return [];
```
Comment (rart) [inline ui/app/src/services/configuration.ts]: Incorrect.

In fact, for now, if the file is missing, while we have a UM, return a mockup with the typical values.

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 22]
```diff
@@ -84,33 +88,59 @@ const useExplorerStyles = makeStyles(() =>
...
 
-const DashboardLink = () => (
-  <LinkWithIcon label="Dashboard" icon="fa fa-tasks" href="/studio/site-dashboard" />
-);
+function SiteDashboardLink() {
+  const { formatMessage } = useIntl();
+  return (
+    <LinkWithIcon
+      link="/studio/dashboard"
+      label={formatMessage(translations['dashboard'])}
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: use dot notation

File: ui/app/src/modules/Preview/Tools/SiteExplorer.tsx [hunk 23]
```diff
@@ -84,33 +88,59 @@ const useExplorerStyles = makeStyles(() =>
...
 
-const SiteConfigLink = () => (
-  <LinkWithIcon label="Site Config" icon="fa fa-sliders" href="/studio/site-config" />
-);
+function SiteConfigLink() {
+  const { formatMessage } = useIntl();
+  return (
+    <LinkWithIcon
+      link="/studio/site-config"
+      label={formatMessage(translations['siteConfig'])}
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/SiteExplorer.tsx]: use dot notation

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@aachavarria please fix conflicts.

**Review by rart — APPROVED** [review]

**Pull request merged**